pkg/controller/common/containerconfig.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"fmt"
+
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GenerateContainerConfig(baseMC *mcfgv1.MachineConfig, cc *mcfgv1.ControllerConfig) (*mcfgv1.MachineConfig, error) {
+	// Create deep copy to avoid mutating input
+	containerMC := &mcfgv1.MachineConfig{
+		Spec: baseMC.Spec, // Copy only the spec we need
+	}
+
+	// Empty ignition config
+	containerMC.Spec.Config = runtime.RawExtension{Raw: []byte("{}")}
+
+	// Generate deterministic name
+	hash, err := HashMachineConfigSpec(containerMC.Spec)
+	if err != nil {
+		return nil, fmt.Errorf("error hashing MachineConfigSpec: %w", err)
+	}
+	containerMC.Name = fmt.Sprintf("container-config-%s", hash)
+
+	// Initialize annotations map if nil
```

Comment:
suggestion (non-blocking): The annotations map will always be nil here since it was just created it above. Instead, let's combine the map initialization with setting the annotations like this:

```golang
containerMC.Annotations = map[string]string{
	ContainerBuildAnnotationKey: "true"
	GeneratedByControllerVersionAnnotationKey: version.Hash,
	ReleaseImageVersionAnnotationKey: cc.Annotations[ReleaseImageVersionAnnotationKey],
}